modules/tinymce/src/core/main/ts/content/GetContentImpl.ts

```diff
@@ -22,9 +22,17 @@ const getContentFromBody = (editor: Editor, args: GetContentArgs, body: HTMLElem
   if (args.format === 'raw') {
     content = Tools.trim(TrimHtml.trimExternal(editor.serializer, body.innerHTML));
   } else if (args.format === 'text') {
-    // return empty string for text format when editor is empty to avoid bogus elements being returned in content
-    content = Zwsp.trim(body.innerText);
-    content = content === '\n' ? '' : content;
+    const doc = editor.getDoc();
```

Comment:
I wonder if we should move this out to a separate function to make it a little more readable?